server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -351,6 +352,41 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
         return new SortFieldAndFormat(field, fieldType.docValueFormat(null, null));
     }
 
+    public boolean canRewriteToMatchNone() {
+        return nestedSort == null && (missing == null || "_last".equals(missing));
+    }
+
+    /**
+     * Returns whether some values of the given {@link QueryShardContext#getIndexReader()} are within the
+     * primary sort value provided in the <code>rawBottomSortValues</code>.
+     */
+    public boolean isBottomSortShardDisjoint(QueryShardContext context, Object[] rawBottomSortValues) {
+        if (rawBottomSortValues == null || rawBottomSortValues.length == 0) {
+            return false;
+        }
+
+        if (canRewriteToMatchNone() == false) {
+            return false;
+        }
+        MappedFieldType fieldType = context.fieldMapper(fieldName);
+        if (fieldType == null) {
+            // unmapped
+            return false;
+        }
+        if (fieldType.indexOptions() == IndexOptions.NONE) {
+            return false;
+        }
+        Object minValue = order() == SortOrder.DESC ? rawBottomSortValues[0] : null;
+        Object maxValue = order() == SortOrder.DESC ? null : rawBottomSortValues[0];
+        try {
+            MappedFieldType.Relation relation = fieldType.isFieldWithinQuery(context.getIndexReader(), minValue, maxValue,
+                true, true, null, DEFAULT_DATE_TIME_FORMATTER.toDateMathParser(), context);
+            return relation == MappedFieldType.Relation.DISJOINT;
+        } catch (Exception exc) {
+            return false;
```

Comment:
I don't like silently ignoring exceptions here, can we narrow down the exception type or remove the catch block entirely? I worry that this might hide bugs.

Reply:
++, this should not be needed with the new logic that send the original date math parser. I removed it.